doc/index.rst

```diff
@@ -721,6 +744,29 @@ Functions
 
    .. versionadded:: 4.1.0
 
+
+Helper classes
```

Comment:
Wasn't sure where to put this so I made a new section. @AlexWaygood do you have any suggestions?

Reply:
In a way, it's a similar class to `ParamSpecArgs` and `ParamSpecKwargs` -- all three are simple data classes that have no special meaning to a type checker, and just exist to hold information

Reply:
"Metadata classes"?

Reply:
Now that "Doc" is a class intended to be use by normal (non-introspection) users, I don't know if "Helper classes" or "Metadata classes" still makes sense. Maybe we should say "Annotation metadata"?

Reply:
"Annotation metadata" sounds reasonable to me